docs/encyclopedia/temporal-nexus-service-operations.mdx

```diff
@@ -0,0 +1,146 @@
+toc_max_heading_level: 4
+slug: /temporalnexus/services-operations
+keywords:
+  - Nexus Services
+  - Nexus Operations
+  - API contracts
+  - durable execution
+  - Temporal SDK
+tags:
+  - distributed systems
+  - integration
+  - temporal
+---
+
+### Nexus Services {#nexus-services}
+
+A Nexus `Service` is a named collection of arbitrary-duration Nexus `Operations` that provide an API contract suitable for sharing across team boundaries. Nexus Services are registered with a Temporal `Worker` that is listening on the target Namespace and Task Queue for an Endpoint.
+
+For example a Nexus Service is often registered in the same Worker as the underlying Workflow they abstract:
+
+```
+func main() {
+	c, err := client.Dial(client.Options{})
+	if err != nil {
+		log.Fatalln("Unable to create client", err)
+	}
+	defer c.Close()
+
+	w := worker.New(c, taskQueue, worker.Options{})
+	service := nexus.NewService(service.HelloServiceName)
+	err = service.Register(handler.EchoOperation, handler.HelloOperation)
+	if err != nil {
+		log.Fatalln("Unable to register operations", err)
+	}
+	w.RegisterNexusService(service)
+	w.RegisterWorkflow(handler.HelloHandlerWorkflow)
+
+	err = w.Run(worker.InterruptCh())
+	if err != nil {
+		log.Fatalln("Unable to start worker", err)
+	}
+}
+```
+
+### Nexus Operations {#nexus-operations}
+
+Nexus `Operations` are not intrinsically durable, but can be made durable by backing them with Temporal primitives like a `Workflow`, `Update`, or `Signal`. The Temporal SDK provides helper functions to create Operations that may be registered with a Service in a Temporal Worker.
+
+Calling a Nexus `Operation` from a caller `Workflow` presents as a non-blocking, arbitrary duration operation. This schedules the `Operation` with the caller’s Temporal cluster and does not block caller `Workflow` execution. The caller’s Nexus Machinery is responsible for making the underlying Nexus RPC calls and updating the caller’s Workflow history with the result.
+
+#### Operation Lifecycle
+
+The Temporal Go SDK is used for the operation lifecycle examples below.
+
+Handlers are created with the Go SDK using:
+
+- `temporalnexus.NewWorkflowRunOperation()`
+  - start a `Workflow` as an asynchronous operation
+- `temporalnexus.NewSyncOperation()`
+  - invoke an underlying `Query` or `Signal` as a synchronous operation
+  - invoke an `Update` as a synchronous operation
+  - execute arbitrary code as a synchronous operation
+
+Caller Workflows use the following to execute a Nexus Operation:
+
+- `svc := Workflow.CreateNexusClient(endpoint, service)` and then
+- `svc.ExecuteOperation()`
+
+##### Asynchronous Operation Lifecycle {#asynchronous-operation-lifecycle}
+
+An asynchronous Nexus Operation may take up to 2 months to complete, to allow for callback URL token validation with asymmetric key rotation at those intervals.
```

Comment:
nit: the 2 month limit is only enforced in Cloud today.

Reply:
Wonder if we should expand on this and explain ScheduleToCloseTimeout.

Reply:
Can explain ScheduleToCloseTimeout in another iteration.